sprint02/api/movies-seeding-and-endpoint-to-get-movies-data

AfaqAhmed-XGrid opened the pull request:
Addresses #6

AfaqAhmed-XGrid committed:
created movie seeding function in db and endpoint
to get movies data with filtering, sorting, searching
and pagination features

Added sorting and filtering functionality to api controller

Added sorting and filtering functionalities in the api controller function.

Also resolved a minor issue in auth.route filtering (middleware was not applied correctly there)

waqar-360 commented on api/modules/movie/movie.controller.js:
```diff
@@ -0,0 +1,66 @@
+  const sort = req.query.sort ? req.query.sort.split(',').join(' ') : ('-releaseDate');
+
+  const [err, movies] = await to(Movie.find({ ...searchQuery, ...queryObj }).sort(sort).skip(skip).limit(limit));
+  const [movieCountErr, movieCount] = await to(Movie.count(searchQuery));
```
can you please check Movie.aggerate() method and use only one query for the movies and count.
instead of making two separate DB calls.

AfaqAhmed-XGrid replied:
Done

waqar-360 commented on api/modules/movie/movie.controller.js:
```diff
@@ -0,0 +1,66 @@
+  logger.info(' Started getting movies data');
+
+  // Filtering
+  const queryObj = { ...req.query };
```
please create separate file or class for pagination. just call the funcation from there.

AfaqAhmed-XGrid replied:
Done

AfaqAhmed-XGrid committed:
used mongodb aggregation

used mongodb aggregate to get data from db with pipeline consisting of filtering, sorting, limit, skip, pageNo, and searching.
Created separate functions to conceive queries from url

waqar-360 commented:
LGTM

AfaqAhmed-XGrid committed:
changed constructFilterQueryObj function

to handel mulptiple filter related queries
